sendto is declared as coming from libc on Windows instead of ws2_32

### Steps to Reproduce

bug.zig:
```zig
const std = @import("std");

pub fn main() !void {
    const msg = &[_]u8 { 0 };
    const send_addr = std.net.Address.parseIp4("255.255.255.255", 1234) catch unreachable;    
    _ = try std.os.sendto(std.os.windows.ws2_32.INVALID_SOCKET, msg, 0, &send_addr.any, send_addr.getOsSockLen());
}
```
```batch
> zig build-exe bug.zig
D:\tools\zig\0.9.0-dev.1387+6f30c8c09\files\lib\std\os.zig:5260:26: error: dependency on libc must be explicitly specified in the build command
        const rc = system.sendto(sockfd, buf.ptr, buf.len, flags, dest_addr, addrlen);
                         ^
D:\tools\zig\0.9.0-dev.1387+6f30c8c09\files\lib\std\os.zig:5262:37: note: referenced here
            if (rc == windows.ws2_32.SOCKET_ERROR) {
                                    ^
D:\tools\zig\0.9.0-dev.1387+6f30c8c09\files\lib\std\os\windows\ws2_32.zig:1845:51: note: referenced here
pub extern "ws2_32" fn WSAGetLastError() callconv(WINAPI) WinsockError;
                                                  ^
D:\tools\zig\0.9.0-dev.1387+6f30c8c09\files\lib\std\os\windows\ws2_32.zig:1845:59: note: referenced here
pub extern "ws2_32" fn WSAGetLastError() callconv(WINAPI) WinsockError;
                                                          ^
D:\tools\zig\0.9.0-dev.1387+6f30c8c09\files\lib\std\os.zig:5263:39: note: referenced here
                switch (windows.ws2_32.WSAGetLastError()) {
                                      ^
D:\tools\zig\0.9.0-dev.1387+6f30c8c09\files\lib\std\os.zig:5283:49: note: referenced here
                    else => |err| return windows.unexpectedWSAError(err),
                                                ^
D:\tools\zig\0.9.0-dev.1387+6f30c8c09\files\lib\std\os.zig:5258:21: note: referenced here
) SendToError!usize {
```

### Expected Behavior

The example should compile.  This example works on zig version 0.8.1.  I believe this was broken by this commit: 7884d84315f4080bfe429d3e29f7ebedf9f3f6  in the file `lib/std/os.zig` with the following change:

```patch
/// When not linking libc, it is the OS-specific system interface.
 pub const system = if (@hasDecl(root, "os") and root.os != @This())
     root.os.system
-else if (builtin.link_libc)
+else if (builtin.link_libc or is_windows)
    std.c
 else switch (builtin.os.tag) {
```

`std.os.sendto` calls `system.sendto` and this change routed `system.sendto` to `std.c` instead of `std.os.windows` where the correct declaration exists on windows.  So this is probably a problem for alot more functions, but this is the first one I found and noticed.  I don't quite understand why this change was made so I don't know whether the fix is to undo this particular line of the commit or if there's more to what's going on here.

### Actual Behavior

Zig thinks `sendto` is defined in `libc` on Windows instead of `ws2_32` where it actually lives.

--- a/lib/std/os.zig
+++ b/lib/std/os.zig
@@ -5913,66 +5913,64 @@ pub fn sendto(
     dest_addr: ?*const sockaddr,
     addrlen: socklen_t,
 ) SendToError!usize {
+    if (builtin.os.tag == .windows) {
+        switch (windows.ws2_32.sendto(sockfd, buf.ptr, buf.len, flags, dest_addr, addrlen)) {
+            windows.ws2_32.SOCKET_ERROR => switch (windows.ws2_32.WSAGetLastError()) {
+                .WSAEACCES => return error.AccessDenied,
+                .WSAEADDRNOTAVAIL => return error.AddressNotAvailable,
+                .WSAECONNRESET => return error.ConnectionResetByPeer,
+                .WSAEMSGSIZE => return error.MessageTooBig,
+                .WSAENOBUFS => return error.SystemResources,
+                .WSAENOTSOCK => return error.FileDescriptorNotASocket,
+                .WSAEAFNOSUPPORT => return error.AddressFamilyNotSupported,
+                .WSAEDESTADDRREQ => unreachable, // A destination address is required.
+                .WSAEFAULT => unreachable, // The lpBuffers, lpTo, lpOverlapped, lpNumberOfBytesSent, or lpCompletionRoutine parameters are not part of the user address space, or the lpTo parameter is too small.
+                .WSAEHOSTUNREACH => return error.NetworkUnreachable,
+                // TODO: WSAEINPROGRESS, WSAEINTR
+                .WSAEINVAL => unreachable,
+                .WSAENETDOWN => return error.NetworkSubsystemFailed,
+                .WSAENETRESET => return error.ConnectionResetByPeer,
+                .WSAENETUNREACH => return error.NetworkUnreachable,
+                .WSAENOTCONN => return error.SocketNotConnected,
+                .WSAESHUTDOWN => unreachable, // The socket has been shut down; it is not possible to WSASendTo on a socket after shutdown has been invoked with how set to SD_SEND or SD_BOTH.
+                .WSAEWOULDBLOCK => return error.WouldBlock,
+                .WSANOTINITIALISED => unreachable, // A successful WSAStartup call must occur before using this function.
+                else => |err| return windows.unexpectedWSAError(err),
+            },
+            else => |rc| return @intCast(usize, rc),
+        }
+    }
     while (true) {
         const rc = system.sendto(sockfd, buf.ptr, buf.len, flags, dest_addr, addrlen);
-        if (builtin.os.tag == .windows) {
-            if (rc == windows.ws2_32.SOCKET_ERROR) {
-                switch (windows.ws2_32.WSAGetLastError()) {
-                    .WSAEACCES => return error.AccessDenied,
-                    .WSAEADDRNOTAVAIL => return error.AddressNotAvailable,
-                    .WSAECONNRESET => return error.ConnectionResetByPeer,
-                    .WSAEMSGSIZE => return error.MessageTooBig,
-                    .WSAENOBUFS => return error.SystemResources,
-                    .WSAENOTSOCK => return error.FileDescriptorNotASocket,
-                    .WSAEAFNOSUPPORT => return error.AddressFamilyNotSupported,
-                    .WSAEDESTADDRREQ => unreachable, // A destination address is required.
-                    .WSAEFAULT => unreachable, // The lpBuffers, lpTo, lpOverlapped, lpNumberOfBytesSent, or lpCompletionRoutine parameters are not part of the user address space, or the lpTo parameter is too small.
-                    .WSAEHOSTUNREACH => return error.NetworkUnreachable,
-                    // TODO: WSAEINPROGRESS, WSAEINTR
-                    .WSAEINVAL => unreachable,
-                    .WSAENETDOWN => return error.NetworkSubsystemFailed,
-                    .WSAENETRESET => return error.ConnectionResetByPeer,
-                    .WSAENETUNREACH => return error.NetworkUnreachable,
-                    .WSAENOTCONN => return error.SocketNotConnected,
-                    .WSAESHUTDOWN => unreachable, // The socket has been shut down; it is not possible to WSASendTo on a socket after shutdown has been invoked with how set to SD_SEND or SD_BOTH.
-                    .WSAEWOULDBLOCK => return error.WouldBlock,
-                    .WSANOTINITIALISED => unreachable, // A successful WSAStartup call must occur before using this function.
-                    else => |err| return windows.unexpectedWSAError(err),
-                }
-            } else {
-                return @intCast(usize, rc);
-            }
-        } else {
-            switch (errno(rc)) {
-                .SUCCESS => return @intCast(usize, rc),
+        switch (errno(rc)) {
+            .SUCCESS => return @intCast(usize, rc),
 
-                .ACCES => return error.AccessDenied,
-                .AGAIN => return error.WouldBlock,
-                .ALREADY => return error.FastOpenAlreadyInProgress,
-                .BADF => unreachable, // always a race condition
-                .CONNRESET => return error.ConnectionResetByPeer,
-                .DESTADDRREQ => unreachable, // The socket is not connection-mode, and no peer address is set.
-                .FAULT => unreachable, // An invalid user space address was specified for an argument.
-                .INTR => continue,
-                .INVAL => return error.UnreachableAddress,
-                .ISCONN => unreachable, // connection-mode socket was connected already but a recipient was specified
-                .MSGSIZE => return error.MessageTooBig,
-                .NOBUFS => return error.SystemResources,
-                .NOMEM => return error.SystemResources,
-                .NOTSOCK => unreachable, // The file descriptor sockfd does not refer to a socket.
-                .OPNOTSUPP => unreachable, // Some bit in the flags argument is inappropriate for the socket type.
-                .PIPE => return error.BrokenPipe,
-                .AFNOSUPPORT => return error.AddressFamilyNotSupported,
-                .LOOP => return error.SymLinkLoop,
-                .NAMETOOLONG => return error.NameTooLong,
-                .NOENT => return error.FileNotFound,
-                .NOTDIR => return error.NotDir,
-                .HOSTUNREACH => return error.NetworkUnreachable,
-                .NETUNREACH => return error.NetworkUnreachable,
-                .NOTCONN => return error.SocketNotConnected,
-                .NETDOWN => return error.NetworkSubsystemFailed,
-                else => |err| return unexpectedErrno(err),
-            }
+            .ACCES => return error.AccessDenied,
+            .AGAIN => return error.WouldBlock,
+            .ALREADY => return error.FastOpenAlreadyInProgress,
+            .BADF => unreachable, // always a race condition
+            .CONNRESET => return error.ConnectionResetByPeer,
+            .DESTADDRREQ => unreachable, // The socket is not connection-mode, and no peer address is set.
+            .FAULT => unreachable, // An invalid user space address was specified for an argument.
+            .INTR => continue,
+            .INVAL => return error.UnreachableAddress,
+            .ISCONN => unreachable, // connection-mode socket was connected already but a recipient was specified
+            .MSGSIZE => return error.MessageTooBig,
+            .NOBUFS => return error.SystemResources,
+            .NOMEM => return error.SystemResources,
+            .NOTSOCK => unreachable, // The file descriptor sockfd does not refer to a socket.
+            .OPNOTSUPP => unreachable, // Some bit in the flags argument is inappropriate for the socket type.
+            .PIPE => return error.BrokenPipe,
+            .AFNOSUPPORT => return error.AddressFamilyNotSupported,
+            .LOOP => return error.SymLinkLoop,
+            .NAMETOOLONG => return error.NameTooLong,
+            .NOENT => return error.FileNotFound,
+            .NOTDIR => return error.NotDir,
+            .HOSTUNREACH => return error.NetworkUnreachable,
+            .NETUNREACH => return error.NetworkUnreachable,
+            .NOTCONN => return error.SocketNotConnected,
+            .NETDOWN => return error.NetworkSubsystemFailed,
+            else => |err| return unexpectedErrno(err),
         }
     }
 }

